backend/Posts/tests.py

```diff
@@ -20,10 +20,123 @@ def setUp(self):
         #self.post1 = Post.objects.create(title="Loving UNLV", message="I love UNLV", hub="Hub 1")
         #self.post2 = Post.objects.create(title="Loving CS", message="I love CS 135", hub="Hub 2")
 
+        self.edit_user = User.objects.create_user(username='edit_hub_user', password='password')
+        self.edit_user_client = APIClient()
+        self.edit_user_client.force_authenticate(user=self.edit_user)
+
+        self.edit_hub_owner = User.objects.create_user(username='edit_hub_owner', password='password')
+        self.edit_hub_owner_client = APIClient()
+        self.edit_hub_owner_client.force_authenticate(user=self.edit_hub_owner)
+
+        self.edit_hub_mod = User.objects.create_user(username='edit_hub_mod', password='password')
+        self.edit_hub_mod_client = APIClient()
+        self.edit_hub_mod_client.force_authenticate(user=self.edit_hub_mod)
+
+        self.edit_hub = Hub.objects.create(name='edit_hub', description='A hub to test editing', owner=self.edit_hub_owner)
+        self.edit_hub.mods.add(self.edit_hub_mod)
+        self.edit_hub.members.add(self.edit_user, self.edit_hub_mod)
+
+        self.edit_dummy_post = {'title': "edit post title user", 'message': "edit post message user", 'hub_id': self.edit_hub.id}
+
+
         self.dummy_post = {'title': "DUMMY POST", 'message': "THIS IS A DUMMY POST"} #MUST ADD "hub_id" KEY
         self.dummy_post2 = {'title': "DUMMY POST 2", 'message': "THIS IS A SECOND DUMMY POST"} # MUST ADD 'hub_id' key
         self.factory = APIRequestFactory()
         
+    def test_user_edit_their_post(self):
+        """
+        Make sure a user can edit their own posts.
+        """
+        response = self.edit_user_client.post(reverse('post-create'), self.edit_dummy_post, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED, "Post was not created")
+
+        new_data = {'title': "edited title", 'message': "edited message"}
+
+        response = self.edit_user_client.patch(reverse('post-edit', kwargs={'id': response.data['id']}), new_data, format='json')
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, "Post could not be edited by post author")
+        self.assertEqual(response.data['title'], new_data['title'], "Post title did not change")
```

Comment:
did change?

Reply:
huh? you are asserting that the response [title] is equal to new data[title] new data

Reply:
maybe im just misunderstaing the third argument in the assertion

Reply:
Oh I think you mean that I forgot to get the post from the database after the API call to make sure that the post was not changed. I'll fix this.